src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -1720,41 +1706,39 @@ void Instruction::execute() {
     }
     case Opcode::AArch64_FCVTDSr: {  // fcvt dd, sn
       // TODO: Handle NaNs, denorms, and saturation?
-      double out[2] = {static_cast<double>(operands[0].get<float>()), 0.0};
-      results[0] = out;
+      results[0] =
+          RegisterValue(static_cast<double>(operands[0].get<float>()), 256);
```

Comment:
Result gone from 2 element array of double to a single `RegisterValue` of 256 bytes? Or am I interpreting this incorrectly?

Reply:
You are interpreting this correctly. 

Arrays were used previously as a way to construct 256 byte `RegisterValue`'s. First, create an array of size 16 bytes (padded with 0s) within the case statement. At the end of the `execute` function,  each `RegisterValue` would be checked for certain conditions. If the conditions are met, it is zero extended from 16 bytes to 256 bytes as shown below:
``` cpp
if ((destinationRegisters[i].type == RegisterType::VECTOR) && !isSVEData_)
      results[i] = results[i].zeroExtend(16, 256); 
```
My change reduces this process to a single constructor call. In this case, we do not need the array at all.